lib/edges/events/action.ex

```diff
@@ -57,10 +58,17 @@ defmodule Edges.Events.Action do
 
   @doc false
   def changeset(%Action{} = action, attrs) do
+    required_field_values = Application.get_env(:edges, :required_field_values)
+
     action
     |> cast(attrs, [:action, :resource_type, :resource_id])
     |> put_assoc(:source, find_or_create_source(attrs))
     |> validate_required([:action, :resource_type, :resource_id, :source])
+    |> validate_inclusion(:action, required_field_values[:action],
+         [message: "Incorrect value for field action"])
+    |> validate_inclusion(:resource_type, required_field_values[:resource_type],
+         [message: "Incorrect value for field resource_type"])
+    |> validate_change(:resource_id, fn(:resource_id, attrs) ->  validation_uuid(:resource_id, attrs) end)
```

Comment:
This looks good. However we also need to validate the value being passed to the `newEvent` mutation in the `person` parameter. The client will send either a uuid, or an auth0 id (e.g. `^(auth0|google-oauth2|facebook)\|[a-zA-Z0-9]+`). It looks like the `person` value is used to create or look up an entry in the `sources` table. We also saw corrupt data being passed in through this parameter and ending up in `sources`, so we do need to validate this as well.

Reply:
added, could you look one more time